app/src/main/java/com/lastaoutdoor/lasta/viewmodel/MoreInfoScreenViewModel.kt

```diff
@@ -66,11 +69,26 @@ constructor(
         is Response.Loading -> {}
         is Response.Success -> {
           val osmData = response.data!!
-          val updatedActivity = activity.copy(startPosition = osmData.getPosition())
+          var newPos = osmData.getPosition()
 
+          if (newPos == Position(0.0, 0.0)) {
+            newPos =
+                when (activity.activityType) {
+                  ActivityType.HIKING,
+                  ActivityType.BIKING -> {
+                    (osmData as Relation)

```

Comment:
This fix is really clever ! I get it because you explained it orally but can you comment it so that others can see it as clearly as I do ?

Reply:
Yes, thank you for your comment, indeed I should state the changement